File: server/src/main/java/org/tbbtalent/server/service/EducationService.java
package org.tbbtalent.server.service;

import org.tbbtalent.server.model.CandidateEducation;
import org.tbbtalent.server.request.education.CreateEducationRequest;

public interface EducationService {

    CandidateEducation createEducation(CreateEducationRequest request);

    CandidateEducation updateEducation(CreateEducationRequest request);

}

File: ui/candidate-portal/src/app/services/education.service.ts
import { Injectable } from '@angular/core';
import {environment} from "../../environments/environment";
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {CandidateEducation} from "../model/candidate-education";

@Injectable({
  providedIn: 'root'
})
export class EducationService {

  private apiUrl: string = environment.apiUrl + '/education';

  constructor(private http: HttpClient) { }

  createEducation(request): Observable<CandidateEducation> {
    return this.http.post<CandidateEducation>(`${this.apiUrl}`, request);
  }

  updateEducation(request): Observable<CandidateEducation> {
     return this.http.post<CandidateEducation>(`${this.apiUrl}/update`, request);
  }

}
